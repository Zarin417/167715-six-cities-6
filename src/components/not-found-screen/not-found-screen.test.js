import React from "react";
import {render} from "@testing-library/react";
import {createMemoryHistory} from "history";
import {Router} from "react-router";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import NotFoundScreen from "./not-found-screen";

const mockStore = configureStore({});
let history;

describe(`Test NotFoundScreen`, () => {
  beforeEach(() => {
    history = createMemoryHistory();
  });

  it(`NotFoundScreen should render correctly, when user is logged`, () => {
    const store = mockStore({
      USER: {isLoggedIn: true, usersEmail: `test@gmail.com`}
    });
    const {container} = render(
        <Provider store={store}>
          <Router history={history}>
            <NotFoundScreen />
          </Router>
        </Provider>
    );

    expect(container).toMatchSnapshot();
  });


  it(`NotFoundScreen should render correctly, when user is not logged`, () => {
    const store = mockStore({
      USER: {isLoggedIn: false, usersEmail: null}
    });

    const {container} = render(
        <Provider store={store}>
          <Router history={history}>
            <NotFoundScreen />
          </Router>
        </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});

