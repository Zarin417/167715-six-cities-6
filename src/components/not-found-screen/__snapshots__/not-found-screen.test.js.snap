// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test NotFoundScreen NotFoundScreen should render correctly, when user is logged 1`] = `
<div>
  <div
    class="not-found-container"
  >
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <article
      class="content"
    >
      <p>
        404: Page is not found
      </p>
      <p>
        <a
          class="backward-btn"
          href="/"
        >
          To the main page
        </a>
      </p>
    </article>
  </div>
</div>
`;

exports[`Test NotFoundScreen NotFoundScreen should render correctly, when user is not logged 1`] = `
<div>
  <div
    class="not-found-container"
  >
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      4
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <span
      class="particle"
    >
      0
    </span>
    <article
      class="content"
    >
      <p>
        404: Page is not found
      </p>
      <p>
        <a
          class="backward-btn"
          href="/"
        >
          To the main page
        </a>
      </p>
    </article>
  </div>
</div>
`;
